### PR TITLE
test(status): increasing test coverage of status module

### PR DESCRIPTION
**Pull request by winding-lines**

### CODE REVIEW

**Comment by winding-lines** [comment]
@seanmonstar  I am done with adding tests in status.rs. I see that you use helper macros in other files, a simple function seems to do the trick here. Let me know if you want me to change/cleanup the code.

**Comment by seanmonstar** [comment]
Excellent!

File: src/status.rs [hunk 1]
```diff
@@ -638,3 +638,101 @@ impl StatusClass {
...
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use super::StatusCode::*;
+
+    // Check that the following entities are properly inter-connected:
+    //   - numerical code
+    //   - status code
+    //   - default code (for the given status code)
+    fn validate(num:u16, status_code:StatusCode, default_code:StatusCode, reason:Option<&str>) {
```
Comment (seanmonstar) [inline src/status.rs]: preferred style is a space after the colon, such as `num: u16`.

File: src/status.rs [hunk 2]
```diff
@@ -638,3 +638,101 @@ impl StatusClass {
...
+        let actual_code = StatusCode::from_u16(num);
+        if actual_code != status_code  {
```
Comment (seanmonstar) [inline src/status.rs]: These can all be `assert_eq!`, and you'll get a decent error message automatically.

**Comment by GitCop** [comment]
Thanks for contributing! Unfortunately, I'm here to tell you there were the following style issues with your Pull Request:

 * Commit: af831f47f299c910ee2ed0f5ea599f3d41bb9a7b
  * Commits must be in the following format: **%{type}(%{scope}): %{description}**

 * Commit: 4ba3033a49119a6744c932cabf7c104db938f81b
  * Commits must be in the following format: **%{type}(%{scope}): %{description}**

Guidelines are available at https://github.com/hyperium/hyper/blob/master/CONTRIBUTING.md

---
This message was auto-generated by https://gitcop.com

**Comment by seanmonstar** [comment]
Looks like this contains a commit from master. Would you rebase? While
doing so, you could squash the commits into 1, using 'git rebase -i
upstream/master'.

**Comment by winding-lines** [comment]
Yeah, I did a git pull from upstream master and something unexpected happened :-) I did a rebase, let me know if the code/history are in the desired state now.

**Comment by seanmonstar** [comment]
Perfect, thank you!

**Pull request merged**